applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -73,17 +73,15 @@ class OpenHDF5File:
     def __init__(self, parameters: KratosMultiphysics.Parameters, model_part: KratosMultiphysics.ModelPart) -> None:
         self.__model_part = model_part
         self.__parameters = parameters
-        self.__file: 'typing.Optional[KratosHDF5.HDF5File]' = None
+        self.__file = CreateHDF5File(self.__model_part, self.__parameters)
 
     def __enter__(self) -> KratosHDF5.HDF5File:
-        self.__file = CreateHDF5File(self.__model_part, self.__parameters)
         return self.__file
 
     def __exit__(self, exit_type, exit_value, exit_traceback) -> None:
         self.__file.Close()
-        self.__file = None
 
     @property
-    def file(self) -> KratosHDF5.HDF5File:
+    def file(self) -> 'typing.Optional[KratosHDF5.HDF5File]':
```

Comment:
```suggestion
    def file(self) -> typing.Optional[KratosHDF5.HDF5File]:
```

this should work too

or `KratosHDF5.HDF5File | None` from python 3.9 on

Reply:
Nonono only python 3.8 for now. Don't make us upgrade the version again or several people will claim my body in pieces.

Reply:
@roigcarlo I was imagining your facial expression when reading that comment ... Hehee...

Reply:
@roigcarlo don't worry, we'll stitch you back in one piece!
I think it's best to support the officially supported python versions (python3.8 and later, at the moment).

Reply:
That was only meant as FYI 😅

typing is still undergoing iterations and getting improved